service/storage_service.cc

```diff
@@ -2767,7 +2767,7 @@ future<> storage_service::removenode_add_ranges(lw_shared_ptr<dht::range_streame
                 my_new_ranges.emplace_back(x.first);
             }
         }
-        std::unordered_multimap<inet_address, dht::token_range> source_ranges = get_new_source_ranges(keyspace_name, my_new_ranges);
+        std::unordered_multimap<inet_address, dht::token_range> source_ranges = co_await get_new_source_ranges(keyspace_name, my_new_ranges);
```

Comment:
Do we know that the token metadata won't change while we calculate this?

Reply:
Addressed in v3